### PR TITLE
Add co_aberration function and sunpos function bugfix 

### PR DESCRIPTION
**Pull request by TestSubjector**
The sunpos function gave the wrong output of `elong` and `obliquity`, because they were
in degrees from the start and so didn't need to be converted again into degrees. Verified with values from sunpos.pro (IDL AstroLib). The bug was found as it propagated into the output of co_aberration function.

* src/sunpos.jl: Bugfix
* TODO.md: Remove 'co_aberration' from list.
* docs/src/ref.md: Add "co_aberration" entry to the manual.
* src/utils.jl: Add co_aberration entry to "utils".
* src/co_aberration.jl: Contains code of co_aberration function.
* test/util-tests.jl: Include tests for co_aberration.

### CODE REVIEW

File: src/co_aberration.jl [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+function _co_aberration{T<:AbstractFloat}(jd::T, ra::T, dec::T, eps::T)
+    t = (jd -2451545)/36525
+    if isnan(eps)
+        eps0 = ten(23,26,21.448)*3600 - 46.815*t - 0.00058*(t^2) +
+               0.001813*(t^3)
+        eps = sec2rad(eps0 + nutate(jd)[2])
+    end
+    sunlong = T(sunpos(jd, radians=true)[3])
+    e = 0.016708634 - 0.000042037*t - 0.0000001267*(t^2)
+    pe = 102.93735 + 1.71946*t + 0.00046*(t^2)
```
Comment (giordano) [inline src/co_aberration.jl]: Use `@evalpoly` for `e` and `pe`.

File: src/co_aberration.jl [hunk 2]
```diff
@@ -0,0 +1,99 @@
+# This file is a part of AstroLib.jl. License is MIT "Expat".
+
+function _co_aberration{T<:AbstractFloat}(jd::T, ra::T, dec::T, eps::T)
+    t = (jd -2451545)/36525
```
Comment (giordano) [inline src/co_aberration.jl]: Use `J2000` in place of 2451545.  It's a constant defined in `common.jl`

File: src/co_aberration.jl [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    if isnan(eps)
+        eps0 = ten(23,26,21.448)*3600 - 46.815*t - 0.00058*(t^2) +
```
Comment (giordano) [inline src/co_aberration.jl]: Use `@evalpoly`.

File: src/co_aberration.jl [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+               0.001813*(t^3)
+        eps = sec2rad(eps0 + nutate(jd)[2])
+    end
+    sunlong = T(sunpos(jd, radians=true)[3])
```
Comment (giordano) [inline src/co_aberration.jl]: It shouldn't be necessary to convert the result to `T`, hopefully `sunpos` already returns a `T` element, doesn't it?

File: src/co_aberration.jl [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    sa = sind(ra)
+    t1 = (cs*ce*(te*cd - sa*sd) + ca*sd*ss)
+    t2 = (cp*ce*(te*cd - sa*sd) + ca*sd*sp)
+    d_ra = 20.49552*(e*(ca*cp*ce + sa*sp) - ca*cs*ce - sa*ss)/cd
+    d_dec = 20.49552*(e*t2 - t1)
+    return d_ra, d_dec
+end
+
+"""
+
```
Comment (giordano) [inline src/co_aberration.jl]: Something missing here? ;-)

File: src/sunpos.jl [hunk 6]
```diff
@@ -53,9 +53,9 @@ function _sunpos{T<:AbstractFloat}(jd::T, radians::Bool)
     ra = cirrange(atan2(sind(l)*cosd(oblt), cosd(l)), 2pi)
     dec = asin(sind(l)*sind(oblt))
     if radians
-        return ra, dec, longmed, oblt
+        return ra, dec, deg2rad(longmed), deg2rad(oblt)
```
Comment (giordano) [inline src/sunpos.jl]: Thanks!!

File: src/co_aberration.jl [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+The output d_ra is *not* multiplied by cos(dec), so that
+apparent_ra = ra + d_ra/3600.
+
+These formula are from Meeus, Chapters 23.  Accuracy is much better than 1
+arcsecond. The maximum deviation due to annual aberration is 20.49'' and occurs when the
+Earth's velocity is perpendicular to the direction of the star.
+
+This function calls [nutate](@ref), [ten](@ref) and [sunpos](@ref).
+"""
+co_aberration(jd::Real, ra::Real, dec::Real; eps::Real=NaN) =
```
Comment (giordano) [inline src/co_aberration.jl]: Maybe `eps` can be an optional argument, rather than a keyword?

File: src/co_aberration.jl [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    _co_aberration(promote(float(jd), float(ra), float(dec), float(eps))...)
+
+function co_aberration{R<:Real}(jd::AbstractVector{R}, ra::AbstractVector{R},
+                                dec::AbstractVector{R}; eps::Real=NaN)
```
Comment (giordano) [inline src/co_aberration.jl]: Same comment about `eps` as above.

File: src/co_aberration.jl (hunk 2 above)
Comment (TestSubjector) [inline src/co_aberration.jl]: Done. Also converted 1/36525 to `inv(JULIANYEAR*100)` as in sunpos.jl

**Pull request merged**